utils/portalloc/src/data/dto.rs

```diff
@@ -10,9 +10,15 @@ pub struct RangeData {
     expires: u64,
 }
 
+fn default_next() -> u16 {
+    crate::LOW
+}
+
 #[derive(Serialize, Deserialize, Debug, Clone, Default)]
 #[serde(rename_all = "kebab-case")]
 pub struct RootData {
+    #[serde(default = "default_next")]
+    pub next: u16,
```

Comment:
This is the core of the idea - store the pointer around the port range in the state file, to move linearly around it and use the available port range quite optimally.